rust-runtime/aws-smithy-types/src/date_time/format.rs

```diff
@@ -377,6 +377,12 @@ pub(crate) mod rfc3339 {
     // Timezones not supported:
     // Not OK: 1985-04-12T23:20:50-02:00
     pub(crate) fn parse(s: &str) -> Result<DateTime, DateTimeParseError> {
+        let timezone_offset = "-02:00".len() - 1;
+        if matches!(s.chars().rev().nth(timezone_offset), Some('-') | Some('+')) {
```

Comment:
`s` is all ASCII right? Can't we index directly instead of consuming the iterator in linear time?

Reply:
Yes, done. Thanks